src/index.js

```diff
@@ -3,16 +3,27 @@ export default middleware
 
 
 export { runSaga } from './internal/runSaga'
-export { CANCEL } from './internal/proc'
 export { END, eventChannel, channel } from './internal/channel'
 export { buffers } from './internal/buffers'
 export { takeEvery, takeLatest } from './internal/sagaHelpers'
-export { delay } from './internal/utils'
 
+import { CANCEL } from './internal/proc'
 import * as effects from './effects'
 import * as utils from './utils'
 
+export function delay(ms, val=true) {
+  let timeoutId;
+  const promise = new Promise((resolve) => {
+    timeoutId = setTimeout(resolve, ms, val);
```

Comment:
The third argument support isn't cross-browser ( http://stackoverflow.com/a/12657665 ), and would require a polyfill (though core-js that comes with babel seems to polyfill it https://github.com/zloirock/core-js/blob/e482646353b489e200a5ecccca6af5c01f0b4ef2/modules/web.timers.js ).